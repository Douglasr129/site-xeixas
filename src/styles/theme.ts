/* eslint-disable quotes */
export const theme = {
	colors: {
		primaryColor: '#0a1128',
		secondaryColor: '#dc143c',
		white: '#FFFFFF',
		mediumGray: '#DDDDDD',
		darkColor: '#2d2d2d',
	},
	font: {
		family: {
			default: "'Open Sans', sans-serif",
			secundary: "'Montserrat',sans-serif",
		},
		sizes: {
			xsmall: '0.8rem',
			small: '1.6rem',
			medium: '2.4rem',
			large: '3.2rem',
			xlarge: '4.0rem',
			xxlarge: '4.8rem',
			huge: '5.6rem',
			xhuge: '6.4rem',
		},
	},
	media: {
		lteMedium: '(max-width: 768px)',
	},
	spacing: {
		xsmall: '0.8rem',
		small: '1.6rem',
		medium: '2.4rem',
		large: '3.2rem',
		xlarge: '4.0rem',
		xxlarge: '4.8rem',
		huge: '5.6rem',
		xhuge: '6.4rem',
	},
};
