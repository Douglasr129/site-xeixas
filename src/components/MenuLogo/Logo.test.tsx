import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { Logo } from '.';

describe('<Logo />', () => {
	it('should render a Logo with link', () => {
		renderTheme(
			<Logo
				srcImg="static/media/public/assets/images/logo.svg"
				text="logo"
				link="/logo"
			/>,
		);
		expect(screen.getByRole('link')).toHaveAttribute('href', '/logo');
	});
	it('should render a Logo with image', () => {
		renderTheme(
			<Logo
				srcImg="static/media/public/assets/images/logo.svg"
				text="logo"
				link="/logo"
			/>,
		);
		expect(screen.getByRole('img')).toHaveAttribute(
			'src',
			'static/media/public/assets/images/logo.svg',
		);
		expect(screen.getByRole('img')).toHaveAttribute('alt', 'logo');
	});
	it('should render a Logo with Text', () => {
		renderTheme(<Logo text="logo" link="/logo" />);
		expect(screen.getByRole('link')).toHaveTextContent('logo');
	});
	it('should render in dark mode', () => {
		const { container } = renderTheme(
			<Logo
				srcImg="static/media/public/assets/images/logo.svg"
				text="logo"
				link="/logo"
			/>,
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});
