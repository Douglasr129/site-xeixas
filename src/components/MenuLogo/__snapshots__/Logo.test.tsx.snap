// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render in dark mode 1`] = `
.c0 {
  display: flex;
  align-items: center;
  text-decoration: none;
  font-family: 'Montserrat',sans-serif;
  font-size: 4.0rem;
  color: inherit;
}

.c0 >img {
  max-height: 3rem;
}

<a
  class="c0"
  href="/logo"
>
  <img
    alt="logo"
    data-nimg="1"
    decoding="async"
    height="160"
    loading="lazy"
    src="static/media/public/assets/images/logo.svg"
    style="color: transparent;"
    width="160"
  />
</a>
`;
